File: gateway/gateway-api/src/main/java/ru/gilko/gatewayapi/dto/rental/RentalDto.java
package ru.gilko.gatewayapi.dto.rental;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import ru.gilko.gatewayapi.dto.car.CarBaseDto;
import ru.gilko.gatewayapi.dto.payment.PaymentDto;
import ru.gilko.rentalapi.constants.enums.RentalStatus;

import java.time.LocalDate;
import java.util.UUID;

@Data
public class RentalDto {
    @Schema(description = "ID аренды")
    private UUID rentalUid;

    @Schema(description = "Статус аренды. В процессе/завершена/отменена.")
    private RentalStatus status;

    @Schema(description = "Дата начала аренды")
    private LocalDate dateFrom;

    @Schema(description = "Дата завершения аренды")
    private LocalDate dateTo;

    @Schema(description = "Арендованный автомобиль")
    private CarBaseDto car;

    @Schema(description = "Информация по оплате")
    private PaymentDto payment;
}

File: frontend/src/dtos/RentalDto.js
import CarRentalDto from "./CarRentalDto";
import PaymentRentalDto from "./PaymentRentalDto";
import RentStatusEnum from "../enums/RentStatusEnum";

class RentalDto {
    rentalUid: String;
    status: RentStatusEnum;
    dateFrom: Date;
    dateTo: Date;

    car: CarRentalDto;
    payment;

    constructor(rentalUid: String,
                status: RentStatusEnum,
                dateFrom: Date,
                dateTo: Date,
                car: CarRentalDto,
                payment: PaymentRentalDto) {
        this.rentalUid = rentalUid;
        this.status = status;
        this.dateFrom = dateFrom;
        this.dateTo = dateTo;
        this.car = car;
        this.payment = payment;
    }
}

export default RentalDto;